lib/src/platform.dart

```diff
@@ -0,0 +1,129 @@
+  static const String flutter = 'flutter';
+  static const String mirrors = 'mirrors';
+
+  static const String angular = 'angular';
+}
+
+class PlatformSummary {
+  final Platform package;
+  final Map<String, Platform> libraries;
+
+  PlatformSummary(this.package, this.libraries);
+
+  bool get hasConflict =>
+      package.hasConflict ||
+      libraries.values.any((p) => p.hasConflict) ||
+      libraries.values.any((p) => p.excludes
+          .any((s) => package.only.contains(s) || package.uses.contains(s)));
+}
+
+class Platform {
+  final List<String> only;
+  final List<String> uses;
+  final List<String> excludes;
+
+  Platform._(this.only, this.uses, this.excludes);
+
+  factory Platform(Iterable<String> only, Iterable<String> uses,
+          Iterable<String> excludes) =>
+      new Platform._(
+        (only ?? []).toList()..sort(),
+        (uses ?? []).toList()..sort(),
+        (excludes ?? []).toList()..sort(),
+      );
+
+  factory Platform.fromMap(Map<String, dynamic> map) =>
+      new Platform(map['only'], map['uses'], map['excludes']);
+
+  bool get isEmpty => only.isEmpty && uses.isEmpty && excludes.isEmpty;
+
+  bool get hasConflict =>
+      only.length > 1 ||
+      excludes.any((s) => only.contains(s) || uses.contains(s));
+
+  Map<String, dynamic> toJson() => <String, dynamic>{
+        'only': only,
+        'uses': uses,
+        'excludes': excludes,
+      };
+}
+
+PlatformSummary classifyPlatforms(
+    Pubspec pubspec, Map<String, List<String>> transientLibs) {
+  Set<String> only = new Set();
+  Set<String> uses = new Set();
+  Set<String> excludes = new Set();
+
+  bool isFlutterOnly = pubspec.hasFlutterKey || pubspec.dependsOnFlutterSdk;
+  if (isFlutterOnly) {
+    only.add(KnownPlatforms.flutter);
+  }
+
+  return new PlatformSummary(
+      new Platform(only, uses, excludes),
+      new Map.fromIterable(transientLibs.keys,
+          value: (key) => classifyPlatform(transientLibs[key])));
+}
+
+Platform classifyPlatform(Iterable<String> dependencies) {
+  Set<String> libs = dependencies.toSet();
+
+  Set<String> only = new Set();
+  Set<String> uses = new Set();
+  Set<String> excludes = new Set();
+
+  bool usesUi = libs.contains('dart:ui');
+  bool usesWeb = _webPackages.any(libs.contains);
+  bool usesIo = libs.contains('dart:io');
+  bool usesMirrors = libs.contains('dart:mirrors');
+
+  // Flutter
+  if (usesUi) {
+    only.add(KnownPlatforms.flutter);
+  }
+  if (usesMirrors) {
+    excludes.add(KnownPlatforms.flutter);
+    uses.add(KnownPlatforms.mirrors);
+  }
+
+  // console
+  if (usesIo && !usesWeb) {
+    uses.add(KnownPlatforms.console);
+    excludes.add(KnownPlatforms.browser);
+  }
+
+  // browser
+  if (usesWeb && !usesIo) {
+    only.add(KnownPlatforms.browser);
+    excludes.add(KnownPlatforms.console);
+  }
+
+  // http and other cross-platform library
+  if (usesWeb && usesIo) {
+    uses.add(KnownPlatforms.browser);
+    uses.add(KnownPlatforms.console);
```

Comment:
the http package is both.

It has one library that is web-only and one library that is io-only.

I thought we were rolling this data up at the library level – then the package can have a mix.